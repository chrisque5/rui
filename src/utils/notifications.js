import { notification, Button } from 'antd';
import { CloseOutlined } from '@ant-design/icons';

const clearAll = () => {
  notification.destroy();
};

const closeIcon = (
  <>
    <Button size="small" data-testid="btnCloseAll" className="close-all" onClick={() => clearAll()}>Close All</Button>
    <CloseOutlined />
  </>
);

const showErrorNotification = (message, description, duration) => {
  notification.error({
    message, description, duration: duration || 5, closeIcon,
  });
};

const showSuccessNotification = (message, description, duration) => {
  notification.success({
    message, description, duration: duration || 0, closeIcon,
  });
};

const showWarningNotification = (message, description, duration) => {
  notification.warning({
    message, description, duration: duration || 5, closeIcon,
  });
};

const showInfoNotification = (message, description, duration) => {
  notification.info({
    message, description, duration: duration || 5, closeIcon,
  });
};

export {
  closeIcon,
  showErrorNotification,
  showSuccessNotification,
  showWarningNotification,
  showInfoNotification,
  clearAll,
};
