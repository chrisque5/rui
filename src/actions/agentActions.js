import { actionTypes } from '../utils/constants';
import { isDealTypeSame } from '../utils/helper';
import { getSelectedDealType } from '../utils/selectors';
import agentApi from '../api/agentApi';

export const loadAgentsSuccess = (agents) => ({ type: actionTypes.LOAD_AGENTDATA_SUCCESS, agents });
export const loadAgentsFailed = (agents) => ({ type: actionTypes.LOAD_AGENTDATA_FAILED, agents });
export const loadAgentsCancelled = () => ({ type: actionTypes.LOAD_AGENTDATA_CANCELLED });
export const resetAgentDataAction = () => ({ type: actionTypes.RESET_AGENTDATA });

export const loadAgents = (dealType) => (dispatch, getState) => (
  agentApi.getAgents(dealType)
    .then((response) => (
      isDealTypeSame(dealType, getSelectedDealType(getState()))
        ? dispatch(loadAgentsSuccess(response))
        : dispatch(loadAgentsCancelled())
    ))
    .catch((error) => dispatch(loadAgentsFailed(error)))
);

export const resetAgentData = () => (dispatch) => dispatch(resetAgentDataAction());
