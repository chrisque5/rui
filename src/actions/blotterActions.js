import { actionTypes } from '../utils/constants';
import blotterApi from '../api/blotterApi';

export const loadRtuBlotterCountSuccess = (payload) => ({ type: actionTypes.LOAD_BLOTTER_RTU_COUNT_SUCCESS, payload });
export const loadRtuBlotterCountFailed = (payload) => ({ type: actionTypes.LOAD_BLOTTER_RTU_COUNT_FAILED, payload });
export const resetRtuBlotterCountSuccess = () => ({ type: actionTypes.RESET_BLOTTER_RTU_COUNT_SUCCESS });
export const loadRtuBlotterCount = () => (dispatch) => (
  blotterApi.getRtuBlotterCount()
    .then((rtuBlotterCount) => {
      dispatch(loadRtuBlotterCountSuccess({ rtuBlotterCount }));
    })
    .catch((error) => dispatch(loadRtuBlotterCountFailed(error)))
);

export const resetRtuBlotterCount = () => (dispatch) => dispatch(resetRtuBlotterCountSuccess());
