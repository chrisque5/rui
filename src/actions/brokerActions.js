import { actionTypes } from '../utils/constants';
import { isDealTypeSame } from '../utils/helper';
import brokerApi from '../api/brokerApi';
import { getSelectedDealType } from '../utils/selectors';

export const loadBrokersSuccess = (brokers) => ({ type: actionTypes.LOAD_BROKERDATA_SUCCESS, brokers });
export const loadBrokersFailed = (error) => ({ type: actionTypes.LOAD_BROKERDATA_FAILED, error });
export const loadBrokersCancelled = () => ({ type: actionTypes.LOAD_BROKERDATA_CANCELLED });
export const loadGcdBrokersSuccess = (gcdBrokers) => ({ type: actionTypes.LOAD_GCD_BROKERDATA_SUCCESS, gcdBrokers });
export const loadGcdBrokersFailed = (error) => ({ type: actionTypes.LOAD_GCD_BROKERDATA_FAILED, error });
export const updateBrokersSuccess = (brokerUpdateStatus) => ({ type: actionTypes.UPDATE_BROKER_SUCCESS, brokerUpdateStatus });
export const updateBrokersFailed = (error) => ({ type: actionTypes.UPDATE_BROKER_FAILED, error });
export const resetBrokerDataAction = () => ({ type: actionTypes.RESET_BROKERDATA });

export const loadBrokers = (dealType, deskId) => (dispatch, getState) => (
  brokerApi.getBrokers(dealType, deskId)
    .then((response) => (
      isDealTypeSame(dealType, getSelectedDealType(getState()))
        ? dispatch(loadBrokersSuccess(response))
        : dispatch(loadBrokersCancelled())
    ))
    .catch((error) => dispatch(loadBrokersFailed(error)))
);

export const loadGcdBrokers = (dealType, deskId) => (dispatch) => (
  brokerApi.getBrokers(dealType, deskId, true)
    .then((response) => dispatch(loadGcdBrokersSuccess(response)))
    .catch((error) => dispatch(loadGcdBrokersFailed(error)))
);

export const updateBrokers = (formValues) => (dispatch) => (
  brokerApi.updateBrokers(formValues)
    .then((response) => dispatch(updateBrokersSuccess(response)))
    .catch((error) => dispatch(updateBrokersFailed(error)))
);

export const loadDealBrokers = (dealType) => (dispatch, getState) => (
  brokerApi.getDealBrokers(dealType)
    .then((response) => (
      isDealTypeSame(dealType, getSelectedDealType(getState()))
        ? dispatch(loadBrokersSuccess(response))
        : dispatch(loadBrokersCancelled())
    ))
    .catch((error) => dispatch(loadBrokersFailed(error)))
);

export const resetBrokerData = () => (dispatch) => dispatch(resetBrokerDataAction());
