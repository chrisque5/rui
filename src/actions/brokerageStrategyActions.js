import { actionTypes } from '../utils/constants';
import brokerageStrategyApi from '../api/brokerageStrategyApi';
import { isDealTypeSame, isStrategyTypeSame } from '../utils/helper';
import { getSelectedDealType, getSelectedStrategyType } from '../utils/selectors';

export const loadBrokerageStrategiesSuccess = (brokerageStrategies) => ({ type: actionTypes.LOAD_BROKERAGE_STRATEGIES_SUCCESS, brokerageStrategies });
export const loadBrokerageStrategiesFailed = (error) => ({ type: actionTypes.LOAD_BROKERAGE_STRATEGIES_FAILED, error });
export const loadBrokerageStrategiesCancelled = () => ({ type: actionTypes.LOAD_BROKERAGE_STRATEGIES_CANCELLED });

export const loadBrokerageStrategies = (dealType, strategyType) => (dispatch, getState) => (
  brokerageStrategyApi.getBrokerageStrategyData(dealType, strategyType)
    .then((response) => ((
      isDealTypeSame(dealType, getSelectedDealType(getState()))
      && isStrategyTypeSame(strategyType, getSelectedStrategyType(getState()))
    ) ? dispatch(loadBrokerageStrategiesSuccess(response))
      : dispatch(loadBrokerageStrategiesCancelled())
    ))
    .catch((error) => dispatch(loadBrokerageStrategiesFailed(error)))
);
