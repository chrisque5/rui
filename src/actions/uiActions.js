import { actionTypes } from '../utils/constants';

export function changeCounterPartySelectionAction(selectedRow) {
  return { type: actionTypes.UPDATE_COUNTERPARTY_SELECTION, selectedRow };
}

export function changeCounterPartySelection(selectedRow) {
  return (dispatch) => dispatch(changeCounterPartySelectionAction(selectedRow));
}

export function changeSelectedPreferenceBrokerAction(payload) {
  return { type: actionTypes.UPDATE_PREFERENCE_BROKER_SELECTION, payload };
}

export function changeSelectedPreferenceBroker(brokerDeskKey) {
  return (dispatch, getState) => {
    const state = getState();
    const { user: { preferences: { preferredBrokers } = {} } = {} } = state;
    const [deskId, brokerId] = brokerDeskKey.split(',');

    const payload = {
      preferredBrokers, brokerId, deskId,
    };

    dispatch(changeSelectedPreferenceBrokerAction(payload));
  };
}

export function changeLastTermSelectionAction(selectedTermId) {
  return { type: actionTypes.UPDATE_LAST_TERM_SELECTION, selectedTermId };
}

export function changeLastTermSelection(selectedTermId) {
  return (dispatch) => dispatch(changeLastTermSelectionAction(selectedTermId));
}

export function changeTermValuesAction(terms) {
  return { type: actionTypes.CHANGE_TERMS, terms };
}

export function changeTermValues(terms) {
  return (dispatch) => dispatch(changeTermValuesAction(terms));
}

export function changeStrategyTypeAction(strategy) {
  return { type: actionTypes.CHANGE_STRATEGY, strategy };
}

export function changeStrategyType(strategy) {
  return (dispatch) => dispatch(changeStrategyTypeAction(strategy));
}

export function changeDealTypeAction(dealType) {
  return { type: actionTypes.CHANGE_DEALTYPE, dealType };
}

export function changeDealType(dealType) {
  return (dispatch) => dispatch(changeDealTypeAction(dealType));
}

export function changePageAction(currentPage) {
  return { type: actionTypes.CHANGE_PAGE, currentPage };
}

export function changePage(currentPage) {
  return (dispatch) => dispatch(changePageAction(currentPage));
}

export function resetUiStateAction() {
  return { type: actionTypes.RESET_UI_STATE };
}

export function resetUiState() {
  return (dispatch) => dispatch(resetUiStateAction());
}

export function changeClientHoverDataSuccess(selectedClientData, clients) {
  return {
    type: actionTypes.CHANGE_CLIENT_HOVER_DATA_SUCCESS,
    selectedClientData,
    clients,
  };
}

export function changeClientHoverData(selectedClientData) {
  return (dispatch, getState) => {
    const { clients } = getState();
    dispatch(changeClientHoverDataSuccess(selectedClientData, clients));
  };
}

export function toggleThirdCPAction(value) {
  return { type: actionTypes.TOGGLE_THIRD_CP, value };
}

export function toggleThirdCP(value) {
  return (dispatch) => dispatch(toggleThirdCPAction(value));
}

export function toggleSettingsModalAction() {
  return { type: actionTypes.TOGGLE_SETTINGS_MODAL };
}

export function toggleSettingsModal() {
  return (dispatch) => dispatch(toggleSettingsModalAction());
}

export function toggleRenameFavoritesPopupAction(selectedFavItem) {
  return { type: actionTypes.TOGGLE_RENAME_FAVORITES_POPUP, selectedFavItem };
}

export function toggleRenameFavoritesPopup(selectedFavItem) {
  return (dispatch) => dispatch(toggleRenameFavoritesPopupAction(selectedFavItem));
}

export function toggleChangeFavoritesColourPopupAction(selectedFavItem) {
  return { type: actionTypes.TOGGLE_CHANGE_FAVORITES_COLOUR, selectedFavItem };
}

export function toggleChangeFavoritesColourPopup(selectedFavItem) {
  return (dispatch) => dispatch(toggleChangeFavoritesColourPopupAction(selectedFavItem));
}

export function enableInterestAction(value) {
  return { type: actionTypes.ENABLE_INTEREST, value };
}

export function enableInterest(value) {
  return (dispatch) => dispatch(enableInterestAction(value));
}

export function toggleTradeDateSubmitPopupAction(isVisible) {
  return { type: actionTypes.TOGGLE_TRADE_DATE_CONFIRM_POPUP, payload: { isVisible } };
}

export function toggleTradeDateSubmitPopup(isVisible) {
  return (dispatch) => dispatch(toggleTradeDateSubmitPopupAction(isVisible));
}

export function toggleIsTradeDateEnabledAction(isTradeDateEnabled) {
  return { type: actionTypes.TOGGLE_IS_TRADE_DATE_ENABLED, payload: { isTradeDateEnabled } };
}

export function toggleIsTradeDateEnabled(isTradeDateEnabled) {
  return (dispatch) => dispatch(toggleIsTradeDateEnabledAction(isTradeDateEnabled));
}

export function updateBlotterGridDetails(blotterGridUpdates) {
  return {
    type: actionTypes.UPDATE_BLOTTER_GRID_DETAILS,
    blotterGridUpdates,
  };
}

export function toggleIsClsOverrideAction(isClsOverride) {
  return { type: actionTypes.TOGGLE_IS_CLS_OVERRIDE, payload: { isClsOverride } };
}

export function toggleIsClsOverride(isClsOverride) {
  return (dispatch) => dispatch(toggleIsClsOverrideAction(isClsOverride));
}

export function resetAdminTabChangesAction(reset) {
  return { type: actionTypes.RESET_ADMIN_TAB_CHANGES, reset };
}

export function resetAdminTabChanges(reset) {
  return (dispatch) => dispatch(resetAdminTabChangesAction(reset));
}

export function enableBrowserForceRefreshAction(isBrowserForceRefreshEnabled) {
  return { type: actionTypes.ENABLE_BROWSER_FORCE_REFRESH, isBrowserForceRefreshEnabled };
}

export function enableBrowserForceRefresh(isBrowserForceRefreshEnabled) {
  return (dispatch) => dispatch(enableBrowserForceRefreshAction(isBrowserForceRefreshEnabled));
}

export function resetBlotterSearchAction(reset) {
  return { type: actionTypes.RESET_BLOTTER_SEARCH, reset };
}

export function resetBlotterSearch(reset) {
  return (dispatch) => dispatch(resetBlotterSearchAction(reset));
}

export function changeDealEditStatusAction(isDealEditInProgress) {
  return { type: actionTypes.CHANGE_DEAL_EDIT_STATUS, isDealEditInProgress };
}

export function changeDealEditStatus(isDealEditInProgress) {
  return (dispatch) => dispatch(changeDealEditStatusAction(isDealEditInProgress));
}

export function changeReAllocationAction(reAllocationSide) {
  return { type: actionTypes.CHANGE_RE_ALLOCATION_SIDE, reAllocationSide };
}

export function changeReAllocation(reAllocationSide) {
  return (dispatch) => dispatch(changeReAllocationAction(reAllocationSide));
}

export function updateDealObjectAction(updatedDealObject) {
  return { type: actionTypes.UPDATE_DEAL_OBJECT, updatedDealObject };
}

export function updateDealObject(updatedDealObject) {
  return (dispatch) => dispatch(updateDealObjectAction(updatedDealObject));
}
