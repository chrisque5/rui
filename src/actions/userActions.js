import { actionTypes } from '../utils/constants';
import userApi from '../api/userApi';

export function loadUserSuccess(user) {
  return { type: actionTypes.LOAD_USER_SUCCESS, user };
}

export function loadUserFailed(user) {
  return { type: actionTypes.LOAD_USER_FAILED, user };
}

export function loadUser() {
  return ((dispatch) => userApi.getUser()
    .then((response) => dispatch(loadUserSuccess(response)))
    .catch((error) => dispatch(loadUserFailed(error)))
  );
}

export function loadPreferencesSuccess(preferences) {
  return { type: actionTypes.LOAD_PREFERENCES_SUCCESS, payload: { ...preferences } };
}

export function loadPreferencesFailed(error) {
  return { type: actionTypes.LOAD_PREFERENCES_FAILED, error };
}

export function loadPreferences() {
  return ((dispatch) => userApi.getPreferences()
    .then((response) => dispatch(loadPreferencesSuccess({ preferences: { ...response }, ...response })))
    .catch((error) => dispatch(loadPreferencesFailed(error)))
  );
}

export function preferencesChangeSuccess(preferences) {
  return { type: actionTypes.CHANGE_PREFERENCE_SUCCESS, payload: { preferences } };
}

export function preferencesChangeFailed(error) {
  return { type: actionTypes.CHANGE_PREFERENCE_FAILED, error };
}

export function deleteBrokerPreferenceSuccess(preferences) {
  return { type: actionTypes.DELETE_BROKER_PREFERENCE_SUCCESS, payload: { ...preferences } };
}

export function deleteBrokerPreferenceFailed(error) {
  return { type: actionTypes.DELETE_BROKER_PREFERENCE_FAILED, error };
}

export function addClientTraderPreferenceSuccess(payload) {
  return { type: actionTypes.ADD_CLIENTTRADER_PREFERENCE_SUCCESS, payload };
}

export function addClientTraderPreferenceFailed(error) {
  return { type: actionTypes.ADD_CLIENTTRADER_PREFERENCE_FAILED, error };
}

export function updateDefaultPreferencesAction(newDeal) {
  return { type: actionTypes.UPDATE_DEFAULTS, newDeal };
}

export function addClientTraderPreference(brokerDeskIds, clientTraderIds, agentId) {
  return async (dispatch, getState) => {
    try {
      const response = await userApi.addClientTraderPreference(brokerDeskIds, clientTraderIds, agentId);

      const state = getState();
      const { ui = {} } = state;
      const { selectedPreferenceFirm = null } = ui;
      const { preferredBrokers = [] } = response;
      const [deskId = null, id = null] = brokerDeskIds;

      const payload = {
        preferences: response, brokerDeskIds, preferredBrokers, brokerId: id, deskId, tradingCustomerId: selectedPreferenceFirm,
      };

      return dispatch(addClientTraderPreferenceSuccess(payload));
    } catch (error) {
      return dispatch(addClientTraderPreferenceFailed(error));
    }
  };
}

export function deleteClientTraderPreference(brokerDeskIds, clientData) {
  return ((dispatch) => userApi.deleteClientTraderPreference(brokerDeskIds, clientData)
    .then((response) => dispatch(preferencesChangeSuccess(response)))
    .catch((error) => dispatch(preferencesChangeFailed(error)))
  );
}

export function editClientTraderPreference(brokerDeskIds, clientTraderId) {
  return ((dispatch) => userApi.editClientTraderPreference(brokerDeskIds, clientTraderId)
    .then((response) => dispatch(preferencesChangeSuccess(response)))
    .catch((error) => dispatch(preferencesChangeFailed(error)))
  );
}

export function deleteBrokerPreference(brokerDeskIds) {
  return ((dispatch) => userApi.deleteBrokerPreference(brokerDeskIds)
    .then((response) => dispatch(deleteBrokerPreferenceSuccess({ preferences: { ...response }, ...response })))
    .catch((error) => dispatch(deleteBrokerPreferenceFailed(error)))
  );
}

export function editBrokerOrder(newOrder, existingOrder, favType, selectedBroker) {
  return ((dispatch) => userApi.editBrokerOrderPreference(
    newOrder, existingOrder, favType, selectedBroker,
  ).then((response) => dispatch(preferencesChangeSuccess(response)))
    .catch((error) => dispatch(preferencesChangeFailed(error)))
  );
}

export function addCurrencyPairPreference(brokerDeskIds, currency1, currency2, dealtCurrency) {
  return ((dispatch) => userApi.addCurrencyPairPreference(brokerDeskIds, currency1, currency2, dealtCurrency)
    .then((response) => dispatch(preferencesChangeSuccess(response)))
    .catch((error) => dispatch(preferencesChangeFailed(error)))
  );
}

export function deleteCurrencyPairPreference(brokerDeskIds, currencyPair) {
  return ((dispatch) => userApi.deleteCurrencyPairPreference(brokerDeskIds, currencyPair)
    .then((response) => dispatch(preferencesChangeSuccess(response)))
    .catch((error) => dispatch(preferencesChangeFailed(error)))
  );
}

export function addTermPreference(brokerDeskIds, term, valueDate) {
  return ((dispatch) => userApi.addTermPreference(brokerDeskIds, term, valueDate)
    .then((response) => dispatch(preferencesChangeSuccess(response)))
    .catch((error) => dispatch(preferencesChangeFailed(error)))
  );
}

export function deleteTermPreference(brokerDeskIds, term) {
  return ((dispatch) => userApi.deleteTermPreference(brokerDeskIds, term)
    .then((response) => dispatch(preferencesChangeSuccess(response)))
    .catch((error) => dispatch(preferencesChangeFailed(error)))
  );
}

export function changePreference(newSettings) {
  return ((dispatch) => userApi.modifyUserSettings(newSettings)
    .then((response) => dispatch(preferencesChangeSuccess(response)))
    .catch((error) => dispatch(preferencesChangeFailed(error)))
  );
}

export function addExecutionVenuePreference(brokerDeskIds, executionVenue) {
  return ((dispatch) => userApi.addExecutionVenuePreference(brokerDeskIds, executionVenue)
    .then((response) => dispatch(preferencesChangeSuccess(response)))
    .catch((error) => dispatch(preferencesChangeFailed(error)))
  );
}

export function deleteExecutionVenuePreference(brokerDeskIds, executionVenue) {
  return ((dispatch) => userApi.deleteExecutionVenuePreference(brokerDeskIds, executionVenue)
    .then((response) => dispatch(preferencesChangeSuccess(response)))
    .catch((error) => dispatch(preferencesChangeFailed(error)))
  );
}

export function updateDefaultPreferences(newDeal) {
  return ((dispatch) => userApi.updateDefaults(newDeal)
    .then((response) => dispatch(preferencesChangeSuccess(response)))
    .catch((error) => dispatch(preferencesChangeFailed(error)))
  );
}

export function initUserData() {
  return (dispatch) => dispatch(loadPreferences()).then(() => dispatch(loadUser()));
}

/**
 * When a firm has been selected, load the clients for that firm
 */
export function selectPreferredFirmComplete(payload) {
  return { type: actionTypes.SELECT_PREFERRED_FIRM_COMPLETE, payload };
}

export function selectPreferredFirm(selectedFirm = {}, brokerDeskIds) {
  return (dispatch, getState) => {
    const state = getState();
    const { user: { preferences: { preferredBrokers } = {} } = {} } = state;
    const [deskId, brokerId] = brokerDeskIds;
    const { tradingCustomerCodeLocationCode } = selectedFirm;

    const payload = {
      preferredBrokers, brokerId, deskId, tradingCustomerCodeLocationCode,
    };

    dispatch(selectPreferredFirmComplete(payload));
  };
}
