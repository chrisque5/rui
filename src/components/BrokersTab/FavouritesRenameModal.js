import { Component } from 'react';
import PropTypes from 'prop-types';
import { Input, Modal } from 'antd';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as _ from 'lodash';
import * as uiActions from '../../actions/uiActions';
import * as userActions from '../../actions/userActions';
import * as notification from '../../utils/notifications';

class FavouritesRenameModal extends Component {
    onValueChange = (e) => {
      this.valueChanged = true;
      this.setState({
        nickName: e.target.value,
      });
    };

    handleSave = () => {
      if (!this.valueChanged) {
        this.closePopup();
      } else if (!this.state.nickName || this.state.nickName.trim() === '') {
        notification.showErrorNotification('Validation Error', 'Blank name cannot be saved');
      } else {
        const favoriteItem = _.clone(this.props.selectedFavItem);
        favoriteItem.nickName = this.state.nickName.trim();
        this.props.userActions.editClientTraderPreference(
          this.props.selectedPreferenceBroker, favoriteItem,
        );
        this.closePopup(this.props.selectedFavItem);
      }
    };

    closePopup = () => {
      this.props.uiActions.toggleRenameFavoritesPopup(this.props.selectedFavItem);
      this.valueChanged = false;
    };

    handleKeyDown = (event) => {
      if (event.key === 'Enter') {
        this.handleSave();
      }
    };

    render() {
      return (
        <Modal
          visible={this.props.visible}
          title="Rename Client/Trader"
          onCancel={this.closePopup}
          okText="Save"
          onOk={this.handleSave}
          bodyStyle={{ padding: 10 }}
          closable
          width={200}
          destroyOnClose
        >
          <div style={{ display: 'inline-flex' }}>
            <span style={{ marginRight: '10px' }}>Client/Trader: </span>
            <div>
              <Input
                size="small"
                name="editFav"
                data-testid="editFav"
                defaultValue={this.props.selectedFavItem ? this.props.selectedFavItem.nickName : ''}
                autoFocus
                onChange={this.onValueChange}
                onKeyDown={(event) => this.handleKeyDown(event)}
                maxLength={50}
              />
            </div>
          </div>
        </Modal>
      );
    }
}

const mapStateToProps = ({ ui }) => ({
  visible: ui.isFavRenamePopupVisible,
  selectedFavItem: ui.selectedFavItem,
  selectedPreferenceBroker: ui.selectedPreferenceBroker,
});

FavouritesRenameModal.propTypes = {
  selectedFavItem: PropTypes.shape().isRequired,
  uiActions: PropTypes.shape().isRequired,
  userActions: PropTypes.shape().isRequired,
  visible: PropTypes.bool.isRequired,
  selectedPreferenceBroker: PropTypes.string.isRequired,
};
const mapDispatchToProps = (dispatch) => ({
  uiActions: bindActionCreators(uiActions, dispatch),
  userActions: bindActionCreators(userActions, dispatch),
});
export default connect(mapStateToProps, mapDispatchToProps)(FavouritesRenameModal);
