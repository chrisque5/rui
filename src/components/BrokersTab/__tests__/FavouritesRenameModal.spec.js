import {
  renderWithProvider, fireEvent, waitFor,
} from 'test-utils/testUtils';
import * as uiActions from '../../../actions/uiActions';
import * as userActions from '../../../actions/userActions';
import FavouritesRenameModal from '../FavouritesRenameModal';
import * as notifications from '../../../utils/notifications';
import { panes } from './FavoriteTestData.json';

describe('<FavouritesRenameModal/>', () => {
  const initialState = {
    ui: {
      isFavRenamePopupVisible: true,
      selectedFavItem: panes[0].clients[0],
      selectedPreferenceBroker: '8158,100114',
    },
  };

  const notificationSpy = jest.spyOn(notifications, 'showErrorNotification');
  const toggleRenameFavoritesPopupSpy = jest.spyOn(uiActions, 'toggleRenameFavoritesPopup');
  const editClientTraderPreferenceSpy = jest.spyOn(userActions, 'editClientTraderPreference');

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders snapshot as expected', () => {
    const { container } = renderWithProvider(
      <FavouritesRenameModal />, { initialState },
    );
    expect(container).toMatchSnapshot();
  });

  test('Verify save withoutempty data change', async () => {
    const { getByTestId, getByText } = renderWithProvider(
      <FavouritesRenameModal />, { initialState },
    );
    const saveBtn = await waitFor(() => getByText('Save'));
    fireEvent.click(saveBtn);
    expect(editClientTraderPreferenceSpy).toHaveBeenCalledTimes(0);
    expect(toggleRenameFavoritesPopupSpy).toHaveBeenCalledTimes(1);

    const input = await waitFor(() => getByTestId('editFav'));
    fireEvent.change(input, { target: { value: '' } });
    fireEvent.click(saveBtn);
    expect(notificationSpy).toHaveBeenCalledTimes(1);
  });

  test('Verify data change and save', async () => {
    const { getByTestId, getByText } = renderWithProvider(
      <FavouritesRenameModal />, { initialState },
    );

    const input = await waitFor(() => getByTestId('editFav'));
    fireEvent.change(input, { target: { value: 'BILL CLINTON RENAMED' } });
    const saveBtn = await waitFor(() => getByText('Save'));
    fireEvent.click(saveBtn);
    expect(editClientTraderPreferenceSpy).toHaveBeenCalledTimes(1);
    expect(toggleRenameFavoritesPopupSpy).toHaveBeenCalledTimes(1);

    editClientTraderPreferenceSpy.mockClear();
    toggleRenameFavoritesPopupSpy.mockClear();

    fireEvent.change(input, { target: { value: 'BILL CLINTON ' } });
    fireEvent.keyDown(input, { key: 'Enter', code: 13 });
    expect(editClientTraderPreferenceSpy).toHaveBeenCalledTimes(1);
    expect(toggleRenameFavoritesPopupSpy).toHaveBeenCalledTimes(1);
  });
});
