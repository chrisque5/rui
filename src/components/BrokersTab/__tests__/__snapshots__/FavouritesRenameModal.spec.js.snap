// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavouritesRenameModal/> renders snapshot as expected 1`] = `<div />`;
